Swapping back from fullscreen moves the window up/left (win32, pyglet 2.1)

This is new in 2.1. I can't reproduce it in 2.0.15.

If you toggle fullscreen and window mode repeatedly the window keeps moving slightly up and to the left. If you keep toggling the window can move outside the desktop area. The step it moves is likely equal to the size of the top bar of the window (and the left window border).

Quick example. Just press `F` a few times and see how the window moves. I don't know if this is an issue on multiple platforms.

```py
import pyglet

window = pyglet.window.Window()
label = pyglet.text.Label('Hello, world!',
                          font_size=36,
                          x=window.width // 2,
                          y=window.height // 2,
                          anchor_x='center',
                          anchor_y='center')

@window.event
def on_draw():
    window.clear()
    label.draw()

@window.event
def on_key_press(symbol, modifiers):
    if symbol == pyglet.window.key.F:
        window.set_fullscreen(not window.fullscreen)

pyglet.app.run()
```

Works 👍 

However I had to bypass a type check in `Options.__setitem__` when setting the audio attribute.

```py
pyglet.options["audio"] = ("openal", "xaudio2", "directsound", "pulse", "silent")
```

```
  File "pyglet\__init__.py", line 260, in __setitem__
    assert type(value).__name__ == self.__annotations__[key], f"Invalid type: '{type(value)}' for '{key}'."
AssertionError: Invalid type: '<class 'tuple'>' for 'audio'.
```

It's looking up the key `'<class 'tuple'>` in annotations, but the only key there is `tuple[str]`. That doesn't work very well. Also the annotation for audio should probably be `tuple[str, ...]` to indicate it's a tuple of varying number of strings.

--- a/pyglet/window/win32/__init__.py
+++ b/pyglet/window/win32/__init__.py
@@ -16,10 +16,10 @@ from pyglet.libs.win32.types import (
     MINMAXINFO,
     RAWINPUT,
     RAWINPUTHEADER,
-    TRACKMOUSEEVENT,
-    WCHAR,
     RECT,
     SIZE,
+    TRACKMOUSEEVENT,
+    WCHAR,
 )
 from pyglet.libs.win32.winkey import chmap, keymap
 
@@ -27,6 +27,7 @@ if compat_platform not in ('cygwin', 'win32'):
     raise ImportError('Not a win32 platform.')  # noqa: EM101
 
 from ctypes import (
+    POINTER,
     byref,
     c_int,
     c_int16,
@@ -38,7 +39,6 @@ from ctypes import (
     memmove,
     sizeof,
     wstring_at,
-    POINTER
 )
 
 import pyglet
@@ -408,11 +408,7 @@ class Win32Window(BaseWindow):
                               constants.SWP_NOOWNERZORDER))
 
     def get_location(self) -> tuple[int, int]:
-        rect = RECT()
-        _user32.GetClientRect(self._hwnd, byref(rect))
-        point = POINT()
-        point.x = rect.left
-        point.y = rect.top
+        point = POINT(0, 0)
         _user32.ClientToScreen(self._hwnd, byref(point))
         return point.x, point.y
 
@@ -783,11 +779,7 @@ class Win32Window(BaseWindow):
         return rect.right - rect.left, rect.bottom - rect.top
 
     def _client_to_window_pos(self, x: int, y: int) -> tuple[int, int]:
-        rect = RECT()
-        rect.left = x
-        rect.top = y
-        _user32.AdjustWindowRectEx(byref(rect),
-            self._ws_style, False, self._ex_ws_style)
+        rect = RECT(x, y, x, y)
 
         if constants.WINDOWS_10_ANNIVERSARY_UPDATE_OR_GREATER:
             _user32.AdjustWindowRectExForDpi(byref(rect),

